lib/FusionInventory/Agent/Task/Inventory/Generic/Ipmi/Fru.pm

```diff
@@ -0,0 +1,55 @@
+#
+# OcsInventory agent - IPMI lan channel report
+#
+# Copyright (c) 2008 Jean Parpaillon <jean.parpaillon@kerlabs.com>
+#
+# The Intelligent Platform Management Interface (IPMI) specification
+# defines a set of common interfaces to a computer system which system
+# administrators can use to monitor system health and manage the
+# system. The IPMI consists of a main controller called the Baseboard
+# Management Controller (BMC) and other satellite controllers.
+#
+package FusionInventory::Agent::Task::Inventory::Generic::Ipmi::Fru;
+
+use strict;
+use warnings;
+
+use FusionInventory::Agent::Tools;
+use FusionInventory::Agent::Tools::Generic;
+
+sub isEnabled {
+    my (%params) = @_;
+    return 0 if $params{no_category}->{psu};
+    return 1;
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+    my $logger    = $params{logger};
+
+    my $fru = getIpmiFru(logger => $logger);
+    my $psu;
+
+    foreach my $descr (keys %$fru) {
+        next unless $descr =~ /^PS(\d+)/;
+
+        $psu = {
+            PARTNUM => $fru->{$descr}->{data}->{'Board Part Number'},
+            SERIAL  => $fru->{$descr}->{data}->{'Board Serial'},
+            POWER   => $fru->{$descr}->{data}->{'Max Power Capacity'},
+            VENDOR  => $fru->{$descr}->{data}->{'Board Mfg'},
+            IS_ATX  => 0,
+        };
+
+        $inventory->addEntry(
```

Comment:
Here there a problem if the powersupply was still added from dmidecode output parsing. You should simply not try ipmitool fru output parsing if the dmidecode module was used. So you should add after the last use in this file the following code:
```
# Only run this module if dmidecode has not been found
our $runMeIfTheseChecksFailed =
    ["FusionInventory::Agent::Task::Inventory::Generic::Dmidecode::Psu"];
```

Reply:
Either of the modules may provide partial data, so i'm merging the data from both on the server. What do you think?

Reply:
I think dmidecode should be kept mandatory before trying ipmitool.
Pushing 2 times the information and leaving the server deciding to merge or not is the wrong way as the inventory could be used for any purpose and this would become an issue then.

Actually we don't have any API to safely merge stuffs. The safe way would only be to agregate the code in one module at the moment. I won't complain if you choose this way.

Reply:
What do we name this new package?